base_layer/core/src/mempool/sync_protocol/initializer.rs

```diff
@@ -103,9 +107,9 @@ impl ServiceInitializer for MempoolSyncInitializer {
                 }
                 log_mdc::extend(mdc.clone());
             }
-
-            MempoolSyncProtocol::new(config, notif_rx, connectivity_event_subscription, mempool)
-                .run()
+            let block_event_stream = base_node.get_block_event_stream();
+            MempoolSyncProtocol::new(config, notif_rx, mempool, connectivity)
+                .run(block_event_stream)
```

Comment:
I think we can use the same pattern throughout
```suggestion
            let block_event_stream = base_node.get_block_event_stream();
            MempoolSyncProtocol::new(config, notif_rx, mempool, connectivity, block_event_stream)
                .run()
```